### PR TITLE
QPlane modes:add/expand on hover learn and recent changes

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

**Comment by kd0aij** [comment]
https://github.com/ArduPilot/ardupilot/pull/12486 prevents hover learning unless a Z controller is running, so I think QHOVER is now the only mode which will allow it.  IIRC it originally worked in QSTABILIZE and QACRO also, but this caused confusion for some users.

Also note that https://github.com/ArduPilot/ardupilot/pull/12480 completely disables the throttle curve for QSTABILIZE and QACRO if Q_THROTTLE_EXPO is zero.  This means that Q_M_THST_HOVER is ignored, and that hover will not necessarily be at mid-stick in those modes.

**Comment by Hwurzburg** [comment]
Mark, the z controller is active in QLOITER isnt it? how does it maintain altitude in QLOITER otherwise?
I will update re throttle expo with a note...

**Comment by kd0aij** [comment]
I forgot about QLOITER... Hover learning will work if the wind is low enough, otherwise the forward motor will be running, or airspeed will be too high, or roll/pitch will be too large.

**Comment by Hwurzburg** [comment]
Same caution applies to QHOVER if station keeping...
I should add hover learn stuff to other modes also....

**Comment by kd0aij** [comment]
Agreed; I test by switching to QHOVER, centering the throttle stick and letting it drift with the wind.

@Hwurzburg Here's a plot showing the throttle curves for several expo values and 2 hover thrust values.  It shows that hover is not at mid-stick with expo=0.  

https://drive.google.com/open?id=1i9zHhhZC9ytrz80HDZB_jj3SZNTX2AOv

File: plane/source/docs/qacro-mode.rst [hunk 1]
```diff
@@ -30,3 +28,11 @@ If vehicle has an airspeed sensor and gain interpolation is desired:
 QACRO flying tips:
 
 Transitions from QACRO mode to any other Q-mode are not aided by the autopilot: no throttle boost is automatically applied. This is critical when airspeed (or high throttle) is needed for attitude control, as with non-vectored dual motor tailsitters. The safest way to transition is to make sure you have sufficient airspeed for elevon authority if flying level, or to establish a stable nose-up hover before switching out of QACRO.
+
+Hover Throttle
+==============
+
+Usually, it is desired to hover in any mode, at mid-stick on throttle, so that transitions between modes is easily accomplished without throttle position changes. This can be adjusted using the :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>` parameter, or automatically learned in QHOVER or QLOITER modes by enabling :ref:`Q_M_HOVER_LEARN<Q_M_HOVER_LEARN>`.
+
+.. note:: If :ref:`Q_THROTTLE_EXPO<Q_THROTTLE_EXPO>` = 0 , then :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>`, whether set manually or learned via Q_M_HOVER_LEARN, is not applied, and the throttle is determined directly from the RC input.
```
Comment (tridge) [inline plane/source/docs/qacro-mode.rst]: add "in QACRO and QSTABILIZE modes" to end of the note

File: plane/source/docs/qhover-mode.rst [hunk 2]
```diff
@@ -121,6 +116,13 @@ Common Problems
    it within an appropriately ventilated enclosure.
 #. Sudden altitude changes caused by light striking the barometer. 
    Assuring sunlight cannot hit the baro will cure this.
+#. QuadPlane slowly descends or climbs until the pilot retakes control in
+   stabilize.  Normally this is caused by not having the throttle stick
+   in the mid position.  This commonly happens when the pilot is
+   switching into an altitude holding mode from a manual flight mode (like QSTABILIZE) on
+   a QuadPlane that does not hover at mid throttle.Usually, it is desired to hover in any mode, at mid-stick on throttle, so that transitions between modes is easily accomplished without throttle position changes. This can be adjusted using the :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>` parameter, or automatically learned in QHOVER or QLOITER modes by enabling :ref:`Q_M_HOVER_LEARN<Q_M_HOVER_LEARN>`.
+
+.. note:: If :ref:`Q_THROTTLE_EXPO<Q_THROTTLE_EXPO>` = 0 , then :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>`, whether set manually or learned via Q_M_HOVER_LEARN, is not applied, and the throttle is determined directly from the RC input.
```
Comment (tridge) [inline plane/source/docs/qhover-mode.rst]: as this is the QHOVER page this is not true

File: plane/source/docs/qloiter-mode.rst [hunk 3]
```diff
@@ -60,6 +60,13 @@ Common Problems
    before take-off is always good and it may help to reduce the
    ``EK2_GLITCH_RAD`` and/or ``EK3_GLITCH_RAD`` parameters (see :ref:`GPS glitch wiki page <gps-failsafe-glitch-protection>`
    for details) to tighten up on the glitch detection.
+#. QuadPlane slowly descends or climbs until the pilot retakes control in
+   stabilize.  Normally this is caused by not having the throttle stick
+   in the mid position.  This commonly happens when the pilot is
+   switching into an altitude holding mode from a manual flight mode (like QSTABILIZE) on
+   a QuadPlane that does not hover at mid throttle.Usually, it is desired to hover in any mode, at mid-stick on throttle, so that transitions between modes is easily accomplished without throttle position changes. This can be adjusted using the :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>` parameter, or automatically learned in QHOVER or QLOITER modes by enabling :ref:`Q_M_HOVER_LEARN<Q_M_HOVER_LEARN>`.
+
+.. note:: If :ref:`Q_THROTTLE_EXPO<Q_THROTTLE_EXPO>` = 0 , then :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>`, whether set manually or learned via Q_M_HOVER_LEARN, is not applied, and the throttle is determined directly from the RC input.
```
Comment (tridge) [inline plane/source/docs/qloiter-mode.rst]: not true in QLOITER

File: plane/source/docs/qstabilize-mode.rst [hunk 4]
```diff
@@ -80,3 +80,10 @@ Common Problems
 -  sudden flips during flight.  This is nearly always caused by
    :ref:`mechanical failures <copter:common-diagnosing-problems-using-logs_mechanical_failures>`
    of the motor or ESCs.
+
+Hover Throttle
+==============
+
+Usually, it is desired to hover in any mode at mid-stick on throttle, so that transitions between modes is easily accomplished without throttle position changes. This can be adjusted using the :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>` parameter, or automatically learned in QHOVER or QLOITER modes by enabling :ref:`Q_M_HOVER_LEARN<Q_M_HOVER_LEARN>`.
+
+.. note:: If :ref:`Q_THROTTLE_EXPO<Q_THROTTLE_EXPO>` = 0 , then :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>`, whether set manually or learned via Q_M_HOVER_LEARN, is not applied, and the throttle is determined directly from the RC input.
```
Comment (tridge) [inline plane/source/docs/qstabilize-mode.rst]: add "in QACRO and QSTABILIZE modes".

File: plane/source/docs/qhover-mode.rst (hunk 2 above)
Comment (Hwurzburg) [reply plane/source/docs/qhover-mode.rst]: done

File: plane/source/docs/qloiter-mode.rst (hunk 3 above)
Comment (Hwurzburg) [reply plane/source/docs/qloiter-mode.rst]: done

Comment (Hwurzburg) [reply plane/source/docs/qloiter-mode.rst]: done

File: plane/source/docs/qstabilize-mode.rst (hunk 4 above)
Comment (Hwurzburg) [reply plane/source/docs/qstabilize-mode.rst]: done

File: plane/source/docs/qhover-mode.rst [hunk 5]
```diff
@@ -121,6 +116,11 @@ Common Problems
    it within an appropriately ventilated enclosure.
 #. Sudden altitude changes caused by light striking the barometer. 
    Assuring sunlight cannot hit the baro will cure this.
+#. QuadPlane slowly descends or climbs until the pilot retakes control in
+   stabilize.  Normally this is caused by not having the throttle stick
+   in the mid position.  This commonly happens when the pilot is
+   switching into an altitude holding mode from a manual flight mode (like QSTABILIZE) on
+   a QuadPlane that does not hover at mid throttle.Usually, it is desired to hover in any mode, at mid-stick on throttle, so that transitions between modes is easily accomplished without throttle position changes. This can be adjusted using the :ref:`Q_M_THST_HOVER<Q_M_THST_HOVER>` parameter, or automatically learned in QHOVER or QLOITER modes by enabling :ref:`Q_M_HOVER_LEARN<Q_M_HOVER_LEARN>`.
```
Comment (kd0aij) [inline plane/source/docs/qhover-mode.rst]: Does the RST parser automatically insert whitespace between sentences? 
s/throttle.Usually/throttle.  Usually/

Comment (Hwurzburg) [reply plane/source/docs/qhover-mode.rst]: no....fixed...thanks

**Pull request merged**